src/expandableCalendar/WeekCalendar/index.tsx

```diff
@@ -149,74 +85,77 @@ class WeekCalendar extends Component<WeekCalendarProps, State> {
         key={item}
         current={item}
         firstDay={firstDay}
-        style={this.getWeekStyle(this.containerWidth, style)}
+        style={getWeekStyle}
         markedDates={markedDates}
         onDayPress={onDayPress}
         context={currentContext}
         numberOfDays={context.numberOfDays}
         timelineLeftInset={context.timelineLeftInset}
       />
     );
-  };
+  },[]);
```

Comment:
No dependencies array?? not even `getWeekStyle`? Also, Let's move the `others` logic out so we won't do it on every item render